python/torch_mlir/extras/onnx_importer.py

```diff
@@ -631,6 +758,318 @@ def tensor_proto_to_attr(self, tp: onnx.TensorProto) -> Attribute:
+    values to an attributes on a node in the interior of a function.
+
+    cf. https://github.com/onnx/onnx/blob/88f8ef15cfaa3138d336f3502aed5018d802bf43/onnx/shape_inference/attribute_binder.h#L15-L64
+    """
+
+    def _bind_attributes_in_subgraph(
+        old_subgraph: onnx.GraphProto,
+        caller_node: onnx.NodeProto,
+        op_schema: onnx.defs.OpSchema,
+    ) -> onnx.GraphProto:
+        """
+        Recurse to bind attributes in a subgraph.
+        """
+        new_subgraph.CopyFrom(old_subgraph)
+        old_nodes = _shallow_copy_and_clear_protobuf_list(new_subgraph.node)
+        for old_node in old_nodes:
+            new_subgraph.node.append(
+                _bind_attributes_on_node(old_node, caller_node, op_schema)
+            )
+        return new_subgraph
+
+    def _bind_attribute(
+        old_attribute: onnx.AttributeProto,
+        caller_node: onnx.NodeProto,
+        op_schema: onnx.defs.OpSchema,
+    ) -> Optional[onnx.AttributeProto]:
+        """
+        Bind a single attribute.
+
+        Bound values either come from attributes on the node calling the
+        function, or from default values. If the attribute is optional and has
+        no default value, and no value was provided by the caller, None is
+        returned and the attribute should be removed.
+        """
+
+        ref_name = old_attribute.ref_attr_name
+        if not ref_name:
+            if not old_attribute.g or len(old_attribute.graphs) == 0:
+                return old_attribute
+
+            # Recurse to bind attributes on subgraphs. ONNX's implementation of
+            # attribute binding only does this for subgraphs that didn't come
+            # from a referenced attribute value, so this code doesn't either.
+            new_attribute = onnx.AttributeProto()
+            new_attribute.CopyFrom(old_attribute)
+            if new_attribute.g:
+                new_attribute.g = _bind_attributes_in_subgraph(
+                    new_attribute.g, caller_node, op_schema
+                )
+            elif new_attribute.graphs:
+                old_subgraphs = _shallow_copy_and_clear_protobuf_list(
+                    new_attribute.graphs
+                )
+                for old_subgraph in old_subgraphs:
+                    new_attribute.graphs.append(
+                        _bind_attributes_in_subgraph(
+                            old_subgraph, caller_node, op_schema
+                        )
+                    )
+            return new_attribute
+
+        for call_attribute in caller_node.attribute:
+            if call_attribute.name == ref_name:
+                new_attribute = onnx.AttributeProto()
+                new_attribute.CopyFrom(call_attribute)
+                new_attribute.name = old_attribute.name
+                return new_attribute
+
+        # The default value is sometimes empty for optional attributes
+        # that don't have a default, in which case it is dropped.
+        if (
```

Comment:
Check about the multi line ifs with spill over. This should be much simpler by request `op_schema.attributes[ref_name].default_value` before the `if` statement entirely and makes the conditional much simpler.

Reply:
Are you suggesting an early-return style, like this?

```python
if not op_schema.attributes[ref_name].default_value:
    return None

if not op_schema.attributes[ref_name].default_value.type:
    return None
```

Reply:
Oh sorry, I see what you mean now.